tests/lib/rules/require-tagless-components.js

```diff
@@ -0,0 +1,124 @@
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/require-tagless-components');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE_REQUIRE_TAGLESS_COMPONENTS: ERROR_MESSAGE } = rule;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: {
+      legacyDecorators: true,
+    },
+  },
+});
+
+ruleTester.run('require-tagless-components', rule, {
+  valid: [
+    `
+      import Component from '@ember/component';
+      export default Component.extend({ tagName: '' });
+    `,
+    `
+      import Component from '@ember/component';
+      export default class MyComponent extends Component {
+        tagName = ''
+      }
+    `,
+    `
+      import Component from '@ember/component';
+      import { tagName } from '@ember-decorators/component';
+      @tagName('')
```

Comment:
I went ahead and decided to support using the decorator, even though it isn't recommended. We are using the decorator heavily in our app, and I'm not sure that our Ember version supports using the `tagName` property instead in ES classes. It would be nice for the rule not to warn about all those classes that are technically not using a wrapper element!